plugins/redfish/fu-redfish-backend.c

```diff
@@ -344,7 +344,8 @@ fu_redfish_backend_coldplug(FuBackend *backend, FuProgress *progress, GError **e
 	    json_object_has_member(json_obj, "MultipartHttpPushUri")) {
 		const gchar *tmp = json_object_get_string_member(json_obj, "MultipartHttpPushUri");
 		if (tmp != NULL) {
-			if (fu_redfish_backend_has_smc_update_path(json_obj)) {
+			if (fu_redfish_backend_has_smc_update_path(json_obj) &&
+			    self->vendor != NULL && g_str_equal(self->vendor, "Supermicro")) {
```

Comment:
Rather than `fu_redfish_backend_has_smc_update_path(json_obj) && self->vendor != NULL && g_str_equal(self->vendor, "Supermicro")` can we please have `g_strcmp0(self->vendor, "Supermicro") == 0 && fu_redfish_backend_has_smc_update_path(json_obj)`

Reply:
Ohh and I think we'll need a fix in redfish.py to set the vendor correctly in this case too.